src/components/PostDetail/components/PostDetailInfo/components/PostDetailStatus/PostDetailStatus.tsx

```diff
@@ -1,7 +1,118 @@
+import { Post, User } from "@/types"
 import * as S from "./PostDetailStatus.Styles"
+import { convertFollowCount } from "@/util/convertFollowCount"
+import PostDetailEditActions from "./components/PostDetailEditActions"
+import ThumbUpOffAltIcon from "@mui/icons-material/ThumbUpOffAlt"
+import ThumbUpAltIcon from "@mui/icons-material/ThumbUpAlt"
+import LinkIcon from "@mui/icons-material/Link"
+import { useMutation, useQueryClient } from "@tanstack/react-query"
+import { likePost, unLikePost } from "@/components/PostDetail/apis/likePost"
+import useModal from "@/components/Modal/hooks/useModal"
+import { useState } from "react"
+import { POST_EDIT_ERROR_MESSAGE } from "@/constants/errorMessage"
+import AlertModal from "@/components/Modal/components/AlertModal/AlertModal"
 
-const PostDetailStatus = () => {
-  return <S.PostDetailStatus></S.PostDetailStatus>
+const MUTATION_KEY = {
+  LIKE_POST_KEY: "IT_IS_LIKE_MUTATION_KEY_546786723746238",
+  UN_LIKE_POST_KEY: "IT_IS_UN_LIKE_MUTATION_KEY_5448718927139",
+}
+
+interface PostDetailStatusProps {
+  post: Post
+  authUser: User
+  isMyPost: boolean
+  isLogin: boolean
+  onClose: () => void
+}
+
+const PostDetailStatus = ({
+  post,
+  authUser,
+  isMyPost,
+  isLogin,
+  onClose,
+}: PostDetailStatusProps) => {
+  const { isShowModal, closeModal, showModal } = useModal()
+  const [alertMessage, setAlertMessage] = useState("")
+
+  const queryClient = useQueryClient()
+  const likeMutate = useMutation({
+    mutationKey: [MUTATION_KEY.LIKE_POST_KEY],
+    mutationFn: likePost,
+    onSuccess: () => {
+      queryClient.refetchQueries()
+    },
+    onError: () => {
+      setAlertMessage(POST_EDIT_ERROR_MESSAGE.LIKE_ERROR)
+      showModal()
+    },
+  })
+  const unlikeMutate = useMutation({
+    mutationKey: [MUTATION_KEY.UN_LIKE_POST_KEY],
+    mutationFn: unLikePost,
+    onSuccess: () => {
+      queryClient.refetchQueries()
+    },
+    onError: () => {
+      setAlertMessage(POST_EDIT_ERROR_MESSAGE.LIKE_ERROR)
+      showModal()
+    },
+  })
+
+  const { likes } = post
+  const myLikePost = post.likes.filter(
+    (likeData) => likeData.user === authUser._id,
+  )[0]
```

Comment:
[P2] 정은님 방법도 좋은데 현재는 post를 찾는 것 같아서 find를 써보는 것은 어떨까요?! 🤔
>find는 제공된 배열에서 제공된 테스트 함수를 만족하는 첫 번째 요소를 반환합니다. 테스트 함수를 만족하는 값이 없으면 `undefined`가 반환됩니다.
```suggestion
  const myLikePost = post.likes.find((likeData) => likeData.user === authUser._id)
```

Reply:
방법이 진짜 많네요 ... 근데 find 좋네요! 바꾸겠슴돵 반드시 좋아요 한 Post 정보를 담고 있어야해서 find가 좋겟네요! 두분다 감사합니다